### PR TITLE
Improve readme for better usability

### PR DESCRIPTION
**Pull request by solidnerd**
This will solve several little issues I stumbled on when I was trying out the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,10 +5,14 @@
 1. Install [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
 2. Install [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) and a [minikube driver](https://github.com/kubernetes/minikube/blob/master/docs/drivers.md). It is recommended to use KVM2 driver for Linux and VirtualBox for MacOS.
 3. Install [kustomize](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/INSTALL.md).
-4. Build the `clusterctl` tool
+4. Clone the Project
+    ```bash
+git clone https://github.com/kubernetes-sigs/cluster-api-provider-azure $(go env GOPATH)/src/sigs.k8s.io/cluster-api-provider-azure
```
Comment (justaugustus) [inline README.md]: These lines need to be indented for the code block to work.

File: README.md [hunk 2]
```diff
@@ -38,12 +43,23 @@ An alternative is to install [Azure CLI](https://docs.microsoft.com/en-us/cli/az
 3. Create the cluster. 
    Kubernetes Version >= 1.11 is required to enable CRD subresources without needing a feature gate.
 
-   ```
-   ./bin/clusterctl create cluster --provider azure \ 
+
+    For Linux use this way:
```
Comment (justaugustus) [inline README.md]:
```suggestion
    **Linux**
```

File: README.md [hunk 3]
```diff
@@ -38,12 +43,23 @@ An alternative is to install [Azure CLI](https://docs.microsoft.com/en-us/cli/az
...
+    ```bash
+    ./bin/clusterctl create cluster --provider azure \
+    -m cmd/clusterctl/examples/azure/out/machines.yaml \
+    -c cmd/clusterctl/examples/azure/out/cluster.yaml \
+    -p cmd/clusterctl/examples/azure/out/provider-components.yaml \
+    --vm-driver kvm2 --minikube kubernetes-version=v1.12.2
+    ```
+
+    For MacOS use this way:
```
Comment (justaugustus) [inline README.md]:
```suggestion
    **macOS** 
```

**Comment by solidnerd** [comment]
@justaugustus done.

**Comment by justaugustus** [comment]
Looks good. Thanks again!
/lgtm
/approve

**Pull request merged**